Helper/Cart.php

```diff
@@ -1833,8 +1835,24 @@ public function createCartByRequest($request)
         //add item to quote
         $item = $request['items'][0];
         $product = $this->productRepository->getbyId($item['reference']);
-        $quote->addProduct($product, $item['quantity']);
-
+        try {
+            $quote->addProduct($product, $item['quantity']);
+        } catch (\Exception $e) {
+            $error_message = $e->getMessage();
+            if ($error_message == 'Product that you are trying to add is not available.') {
```

Comment:
We should not rely on message text but in this case I guess it's OK because there's a fallback on the second expression which is true in both cases. The first one seems more precise if the message matches. LGTM.

Reply:
I also really agree that this is a bad pattern. Preferable if we check the error code instead since that has less of a chance of changing over the message. I want this in 2.3.0 though and it's unlikely to break in one patch - I'll make a backlog task in Asana to strengthen this @vitaliyreznikov